huaweicloud/services/live/resource_huaweicloud_live_domain.go

```diff
@@ -0,0 +1,325 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	v1 "github.com/huaweicloud/huaweicloud-sdk-go-v3/services/live/v1"
+	"github.com/huaweicloud/huaweicloud-sdk-go-v3/services/live/v1/model"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+const (
+	domainTypePull = "pull"
+	domainTypePush = "push"
+
+	domainStatusOn  = "on"
+	domainStatusOff = "off"
+)
+
+func ResourceDomain() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDomainCreate,
+		UpdateContext: resourceDomainUpdate,
+		DeleteContext: resourceDomainDelete,
+		ReadContext:   resourceDomainRead,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{domainTypePull, domainTypePush}, false),
+			},
+
+			"ingest_domain_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"status": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{domainStatusOn, domainStatusOff}, false),
+				Computed:     true,
+			},
+
+			"cname": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceDomainCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcLiveV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Live v1 client: %s", err)
+	}
+
+	createOpts, err := buildCreateParams(d, region)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	log.Printf("[DEBUG] Create Live domain params : %#v", createOpts)
+
+	_, err = client.CreateDomain(createOpts)
+	if err != nil {
+		return diag.Errorf("error creating Live domain: %s", err)
+	}
+
+	d.SetId(createOpts.Body.Domain)
+
+	err = waitingForDomainStatus(ctx, client, d.Id(), model.GetDecoupledLiveDomainInfoStatusEnum().ON,
+		d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// associate the streaming domain name with an ingest domain
+	err = associatingDomain(d, client)
```

Comment:
we can not change the status when creating?

Reply:
If it is a disabled domain, no need to create it.

Reply:
added